### ISSUE TITLE
Disallow re-mapping attributes to existing attributes

### ISSUE THREAD

**Issue by corymhall** [pulumi/pulumi-terraform-bridge#2283]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

```go
map[string]rschema.Attribute{
	"id": StringAttribute{Computed: true},
	"s":  rschema.StringAttribute{Optional: true},
}

info.Resource{
	"some_resource": {
		Tok:       "...",
		ComputeID: computeIDField,
		Fields: map[string]*info.Schema{
			"id": {
                 Name: "s",
             },
		},
	},
}
```

The above is an example of having an attribute that you have re-mapped (via `Schema.Name`) to an existing attribute. Typically the `Schema.Name` is used to rename the attribute, but it is technically possible to create a collision with an existing attribute. This should probably be disallowed.

### Affected area/feature

We should probably add a check that fails during tfgen.

**Comment by iwahbe** [comment]
I agree that this would be a good warning. We would need to do some cycle checking (A renamed to "b", B renamed to "a" is ok), but I think that could be as simple as verifying that the number of unique property names is the same as the number of properties.
